src/toil_scripts/lib/programs.py

```diff
@@ -83,36 +112,73 @@ def docker_call(tool,
     if env:
         for e, v in env.iteritems():
             base_docker_call.extend(['-e', '{}={}'.format(e, v)])
+    
     if docker_parameters:
         base_docker_call += docker_parameters
+   
+    docker_call = []
+    
+    run_pipe = False
 
-    _log.debug("Calling docker with %s." % " ".join(base_docker_call + [tool] + parameters))
-
-    docker_call = base_docker_call + [tool] + parameters
+    if bool(tools) == bool(tool):
+        raise Exception('Either "tool" or "tools" must contain a value, but not both.')
+    if not tools:
+        tools = [ tool ]
+    else:
+        run_pipe = True
+ 
+    # Pipe functionality
+    #   each element in the parameters list must represent a sub-pipe command
+    shell_flag = True      # Flag for running subprocess with string command or list command
+    if run_pipe:
+        command_list = []
+        if len(tools) > 1:
+            # If tool is a list containing multiple docker container name strings
+            #   then format the docker call in the 'pipe-of-containers' mode
+            docker_call.extend(base_docker_call + ['--entrypoint /bin/bash', tools[0], '-c \'{}\''.format(parameters[0])])
+            for i in xrange(1, len(tools)):
+                docker_call.extend(['|'] + base_docker_call + ['-i --entrypoint /bin/bash', tools[i], '-c \'{}\''.format(parameters[i])])
```

Comment:
There should be a check that ensure that the two lists have the same length. Right now the user would get a pretty hard to understand index error if the he/she didn't match the array lengths.